libs/datasets/data_source.py

```diff
@@ -63,3 +69,23 @@ def make_dataset(cls) -> timeseries.MultiRegionDataset:
             )
 
         return MultiRegionDataset.from_fips_timeseries_df(data).add_provenance_all(cls.SOURCE_NAME)
+
+
+# TODO(tom): Move all the ccd_helpers code to this repo
+# TODO(tom): Clean up the mess that is subclasses of DataSource and
+#  instances of DataSourceAndRegionMasks
+class CanScraperBase(DataSource):
+    # The method called to transform the DataFrame returned by CovidCountyDataset into what is
+    # consumed by DataSource.make_dataset. Must be set in subclasses.
+    TRANSFORM_METHOD: Callable[[pd.DataFrame], pd.DataFrame]
+
+    @staticmethod
+    @lru_cache(None)
+    def _get_covid_county_dataset() -> ccd_helpers.CovidCountyDataset:
```

Comment:
Ah - this is probably a todo, but we should rename this to be `CanScraperDataset` or something similar - `CovidCountyDataset` name is a bit outdated

Reply:
'dataset' is pretty overloaded. Added TODO with suggestion  `CanScraperLoader`